File: Sources/Connection.swift
import Foundation

struct Connection: Identifiable {
    enum Status: String, Codable {
        case pending
        case accepted
        case rejected
        case blocked
    }

    let id: UUID
    let requesterUserId: UUID
    let recipientUserId: UUID
    var status: Status = .pending
    let createdAt: Date
    var updatedAt: Date
}

File: models/connection.js
class Connection {
    constructor(id, requesterUserId, recipientUserId, status, createdAt, updatedAt) {
        this.id = id;
        this.requesterUserId = requesterUserId;
        this.recipientUserId = recipientUserId;
        this.status = status;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
}

module.exports = Connection;
